### PR TITLE
improvement: implement new design of timer

### PR DESCRIPTION
**Pull request by brandstetterm**
## Description

In order to implement the new design for our timer, changes to our database and backend had to be made. In order to show the progress bar for the remaining time, we had to save both start- and end-time of the timer to calculate the elapsed time percentage. 

## Changelist:
- Add `timerEnd` to database
- Change backend to save both `timerStart` and `timerEnd`
- Change client redux logic to include `timerStart`
- Add timer utils functions to extract logic out of the components
- Change timer component and styling
- Update tests

## Future tasks
- Implement short action button to add another minute to a running timer

## Visual Changes
Screenshot 2022-12-21 at 13 31 45
Screenshot 2022-12-21 at 13 31 07
Screenshot 2022-12-21 at 13 30 52

### CODE REVIEW

File: src/components/Timer/Timer.scss [hunk 1]
```diff
@@ -1,61 +1,141 @@
...
+  padding-left: $timer__content-height;
+  line-height: $timer__content-height;
   text-align: center;
+  font-weight: bold;
+}
+
+.timer__progress-bar {
+  position: absolute;
+  top: 0;
+  left: 19px;
```
Comment (Resaki1) [inline src/components/Timer/Timer.scss:43]:
```suggestion
  left: calc($timer__content-height / 2);
```

maybe `calc($timer__content-height / 2)` could be a new variable as well, you use it quite often

File: src/components/Timer/Timer.scss [hunk 2]
```diff
@@ -1,61 +1,141 @@
...
+// Short Actions Button Tooltip
+.short-actions__button-wrapper > span {
+  visibility: hidden;
+  opacity: 0;
+  transition: opacity 150ms linear;
+  position: absolute;
+  left: 50%;
+  transform: translateX(-50%);
+  height: 36px;
+  border-radius: 18px;
```
Comment (Resaki1) [inline src/components/Timer/Timer.scss:105]:
```suggestion
  border-radius: calc($timer__content-height / 2);
```

File: src/components/Timer/Timer.scss [hunk 3]
```diff
@@ -1,61 +1,141 @@
...
+  padding: 5px $padding--default;
+  background-color: $tooltip-background-color--light;
+  color: $menu-icon-background-color--dark;
+  font-weight: bold;
+  height: auto;
+  bottom: calc(100% + 4px);
+  box-shadow: 0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1);
```
Comment (Resaki1) [inline src/components/Timer/Timer.scss:112]:
```suggestion
  box-shadow: $box-shadow--light;
```

Comment (Resaki1) [reply src/components/Timer/Timer.scss:112]: and add the --dark version accordingly

File: src/components/Timer/Timer.tsx [hunk 4]
```diff
@@ -107,14 +102,21 @@ export const Timer = (props: TimerProps) => {
...
-          <CloseIcon />
-        </button>
+        <div className="timer__short-actions">
+          <div className="short-actions__button-wrapper">
+            <button onClick={() => store.dispatch(Actions.cancelTimer())}>
+              <CancelIcon />
+            </button>
+            <span>{t("VoteDisplay.finishActionTooltip")}</span>
+          </div>
+        </div>
```
Comment (Resaki1) [inline src/components/Timer/Timer.tsx:117]: are both wrappers needed here?

**Review by Resaki1 — CHANGES_REQUESTED** [review]
To me it feels kinda weird that the progress bar disappears when hovering the timer. Is there a reason why you made it like that?

File: src/components/Timer/Timer.tsx (hunk 4 above)
Comment (brandstetterm) [reply src/components/Timer/Timer.tsx:117]: yes, there will be a second short action soon

**Review by Resaki1 — APPROVED** [review]

**Pull request merged**